homeassistant/components/recorder/migration.py

```diff
@@ -2222,7 +2289,18 @@ def __init__(self, schema_version: int, migration_changes: dict[str, int]) -> No
         self.schema_version = schema_version
         self.migration_changes = migration_changes
 
-    def do_migrate(self, instance: Recorder, session: Session) -> None:
+    def migrate_all(
```

Comment:
Instead of adding this method to the base class, maybe there should be separate base classed for live and non-live data migrators.

Reply:
In isolation, its probably ok to add here but I think you do need to separate anyways to handle the retry case https://github.com/home-assistant/core/pull/125309#discussion_r1783092827